python/semantic_kernel/models/ai/chat_completion/chat_history.py

```diff
@@ -167,6 +170,43 @@ def __eq__(self, other: "ChatHistory") -> bool:
 
         return self.messages == other.messages
 
+    @classmethod
+    def from_rendered_prompt(cls, rendered_prompt: str) -> "ChatHistory":
+        """
+        Create a ChatHistory instance from a rendered prompt.
+
+        Args:
+            rendered_prompt (str): The rendered prompt to convert to a ChatHistory instance.
+
+        Returns:
+            ChatHistory: The ChatHistory instance created from the rendered prompt.
+        """
+        messages: List[ChatMessageContent] = []
+        result, remainder = cls._render_remaining(rendered_prompt, True)
+        if result:
+            messages.append(result)
+        while remainder:
+            result, remainder = cls._render_remaining(remainder)
+            if result:
+                messages.append(result)
+        return cls(messages=messages)
+
+    @staticmethod
+    def _render_remaining(prompt: Optional[str], first: bool = False) -> Tuple[ChatMessageContent, Optional[str]]:
+        """Render the remaining messages in the history."""
+        if not prompt:
+            return None, None
+        prompt = prompt.strip()
+        start = prompt.find("<message")
+        end = prompt.find("</message>")
+        if start == -1 or end == -1:
+            return ChatMessageContent(role=ChatRole.SYSTEM if first else ChatRole.USER, content=prompt), None
+        if start > 0 and end > 0:
+            return ChatMessageContent(role=ChatRole.SYSTEM if first else ChatRole.USER, content=prompt[:start]), prompt[
+                start:
+            ]
+        return ChatMessageContent.from_element(ET.fromstring(prompt[start : end + 10])), prompt[end + 10 :]
```

Comment:
What is the "10" for? is it the length of `</message>`? instead of +10? Can we just do `len(end)` so it can update more seamlessly if we ever change the start/end tags?